upstairs/src/downstairs.rs

```diff
@@ -2524,50 +2405,60 @@ impl Downstairs {
     /// That's what this function provides. To ensure this function is only
     /// used for that purpose, it will panic if [io] is not a repair-related
```

Comment:
We took out the assertion, so the comment here is wrong, but I think you are also
getting rid of this method, and the comment will go with that refactor.

Reply:
I believe the assertion is still present (L2435), so this commit is still internally consistent (but yes, we're removing this method further down the line).